onnxruntime/core/providers/openvino/openvino_execution_provider.cc

```diff
@@ -0,0 +1,571 @@
+  model_proto.set_ir_version(ONNX_NAMESPACE::Version::IR_VERSION);
+
+  return model_proto;
+}
+
+//Checks whether the node is supported by OpenVINO
+bool IsOpSupported(std::string name){
+
+    std::set<std::string> supported_ops = {
+        "Add",
+        "BatchNormalization",
+        "Conv",
+        "GlobalAveragePool",
+        "Relu",
+        "Reshape",
+        "Flatten",
+        "Gemm",
+        "MaxPool",
+        "AveragePool",
+        "Concat",
+        "Dropout",
+        "LRN",
+        "Softmax",
+        "Mul",
+        "Sum",
+        "Transpose",
+        "Identity",
+        "MatMul",
+        "Unsqueeze",
+        "ImageScaler",
+        "LeakyRelu",
+        "GlobalMaxPool"};
+
+    auto iter = supported_ops.find(name);
+    return iter != supported_ops.end();
+}
+
+
+//Checks if the entire graph is supported by OpenVINO EP and returns false if it is not.
+
+bool IsGraphSupported(const onnxruntime::GraphViewer& graph_viewer, std::string dev_id){
+
+  auto initializers = graph_viewer.GetAllInitializedTensors();
+
+  auto node_indexes = graph_viewer.GetNodesInTopologicalOrder();
+
+  auto model_proto = GetModelProtoFromFusedNode(graph_viewer);
+
+  auto graph_proto = model_proto.mutable_graph();
+  int input_dims = 0;
+  int output_dims = 0;
+  int num_inputs = graph_viewer.GetInputs().size();
+  int num_outputs = graph_viewer.GetOutputs().size();
+
+  //input_arrays hold the dimensions of graph inputs
+  std::vector<std::vector<int>> input_arrays;
+
+  if (num_inputs != 0)
+    input_dims = graph_proto->input(0).type().tensor_type().shape().dim_size();
+
+  if (num_outputs != 0)
+    output_dims = graph_proto->output(0).type().tensor_type().shape().dim_size();
+
+  //Get input dimensions
+  if (num_inputs != 0) {
+    for (int i = 0; i < num_inputs; i++) {
+      int input_dims_size = graph_proto->input(i).type().tensor_type().shape().dim_size();
+      std::vector<int> temp_arr;
+
+      for (int j = 0; j < input_dims_size; j++) {
+        temp_arr.push_back(graph_proto->input(i).type().tensor_type().shape().dim(j).dim_value());
+      }
+      input_arrays.push_back(temp_arr);
+    }
+  }
+
+  //GPU Plugin does not support single dimensional input and 5 dimensional input
+  if (dev_id == "GPU") {
+    if (input_dims == 1 || input_dims == 5 || output_dims == 5)
+      return false;
+  }
+
+  bool op_sum = false;
+  bool op_gemm = false;
+
+  for (auto index : node_indexes) {
+    auto node = graph_viewer.GetNode(index);
+
+
+    //Check if the Operation is Supported by OpenVINO
+    if (!IsOpSupported(node->OpType())) {
+      return false;
+    }
+
+    if (node->OpType() == "Sum") {
+      op_sum = true;
+    }
+
+    if (node->OpType() == "Gemm") {
+      op_gemm = true;
+    }
+
+    //BatchNorm, Conv and Reshape cant take more than 1 input
+    if (node->OpType() == "BatchNormalization" || node->OpType() == "Conv" || node->OpType() == "Reshape") {
```

Comment:
what does this mean? conv can't take a bias input?
batchnorm can't take scale/bias inputs?

Reply:
Yes they have to be passed in as initializers

Reply:
Cool I will add more comments

Reply:
In some of the unit tests, biases and weights are passed as inputs.

Reply:
Hey George, 
When i say inputs, i'm not including initializers. So for reshape even though the onnx spec says they are inputs, shape is an initializer in most cases. OpenVINO handles this case.

Only when both of them are inputs, we don't support it. So maybe i should add a comment saying that Reshape must have shape as an initializer